### PR TITLE
Autocomplete tas

### PR DESCRIPTION
**Pull request by afrasier**
* Add `AutocompleteView` which moves all of the autocomplete code into drf view friendly structures
* Add `/accounts/tas/autocomplete/` autocomplete
* Adds the ability to use filters on autocompletes (A nice bonus to moving to AutocompleteView!)
* Adds tests
* Updates docs

### CODE REVIEW

File: usaspending_api/api_docs/markdown/using_the_api.md [hunk 1]
```diff
@@ -50,6 +50,11 @@ The currently available endpoints are listed below. To reduce unnecessary data t
     - _Methods_: GET, POST
 
 
+  * **/v1/accounts/tas/autocomplete/**
```
Comment (bsweger) [inline usaspending_api/api_docs/markdown/using_the_api.md]: Thanks for updating the docs. Do you think this (or one of the other autocomplete endpoints) is worthy of a cookbook entry?

File: usaspending_api/api_docs/markdown/data_dictionary.md [hunk 2]
```diff
@@ -100,6 +100,7 @@ In this section you will find definitions for common terms used in the API. For
 | [/api/v1/awards/total/](/api/v1/awards/total/) | POST |  Aggregate (see [Using the API](/docs/using-the-api)) | Supports aggregation on award records |
 | [/api/v1/accounts/](/api/v1/accounts/) | GET, POST | <a href="#accounts">Appropriation Account Balances</a> | Returns a list of appropriation account balances |
 | [/api/v1/accounts/tas/](/api/v1/accounts/tas/) | GET, POST | <a href="#tas">Treasury Appropriation Account</a> | Returns a list of treasury appropriation accounts, by TAS |
+| [/api/v1/accounts/tas/autocomplete/](/api/v1/accounts/tas/autocomplete/) | POST | Autocomplete (see [Using the API](/docs/using-the-api))| Supports autocomplete on TAS records |
```
Comment (bsweger) [inline usaspending_api/api_docs/markdown/data_dictionary.md]: Am curious--is there a reason that autocompletes can't also get `GET` requests?

File: usaspending_api/references/tests/test_agency_autocomplete.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import json
```
Comment (bsweger) [inline usaspending_api/references/tests/test_agency_autocomplete.py]: Thanks for backfilling these unit tests!

File: usaspending_api/references/views.py [hunk 4]
```diff
@@ -29,15 +29,17 @@ def post(self, request, geocomplete=False, format=None):
         return Response(response_data)
 
 
-class AgencyAutocomplete(APIView):
+class AgencyAutocomplete(FilterQuerysetMixin,
```
Comment (bsweger) [inline usaspending_api/references/views.py]: Thanks for converting to the new views/mixins! That should allow us to hook in to the DRF caching stuff, too.

File: usaspending_api/accounts/tests/test_tas_autocomplete.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+from usaspending_api.accounts.models import TreasuryAppropriationAccount
+from usaspending_api.common.tests.autocomplete import check_autocomplete
+
+
+@pytest.fixture
+def tas_data(db):
+    mommy.make(
+        TreasuryAppropriationAccount,
+        tas_rendering_label='zzz',
```
Comment (bsweger) [inline usaspending_api/accounts/tests/test_tas_autocomplete.py]: Thanks for adding tests, but I would like to point out that your test `tas_rendering_labels` are not nearly as amusing as your fake agency names, which is a little disappointing tbh

File: usaspending_api/api_docs/markdown/data_dictionary.md (hunk 2 above)
Comment (afrasier) [inline usaspending_api/api_docs/markdown/data_dictionary.md]: We'd need to sort out a way to format the get requests, which is out of scope for this but worth a look

File: usaspending_api/accounts/tests/test_tas_autocomplete.py (hunk 5 above)
Comment (afrasier) [inline usaspending_api/accounts/tests/test_tas_autocomplete.py]: I just copy and pasted most of these :P Someone else came up with the names first.

File: usaspending_api/api_docs/markdown/using_the_api.md (hunk 1 above)
Comment (afrasier) [inline usaspending_api/api_docs/markdown/using_the_api.md]: Yes I will include these when I do a big recipe doc update.

**Pull request merged**